### PR TITLE
Screen for a merchant to choose connection to bluetooth or built in reader

### LINKED ISSUE #8134: [Mobile Payments] Add screen for merchant to choose connection to Bluetooth or Built-in reader
- [x] Show to US merchants only ~(verify this firstly, maybe on android we can work worldwide?)~ **- THE US ONKY (will be addressed in another ticket)**
- [x] Show only on supported devices (NFC)

Image

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8134

### Description

If the feature flag is ON, then we show the dialog where a merchant can choose either BT reader or built-in reader

### Testing instructions

With the Feature flag ON
* Start payment flow
* Click CARD
* Notice the new dialog
* Click on "tap to pay". Notice that nothing happens (not implemented yet)
* Click on "BT reader". Notice that the usual IPP flow goes on

With the Feature flag OFF
The current flow should be available only (not reader type picker should be accessible)

### Images/gif

https://user-images.githubusercontent.com/4923871/212274802-71a36ce9-35c0-40a2-b091-80c8ded3dcc2.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <ImageView
+            android:id="@+id/card_reader_type_selection_illustration"
+            android:layout_width="105dp"
+            android:layout_height="85dp"
+            android:layout_gravity="center"
+            android:layout_marginTop="@dimen/margin_extra_extra_medium_large"
+            android:contentDescription="@null"
+            app:srcCompat="@drawable/img_ipp_reader_type_selection"
```
Comment (kidinov) [inline WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml:40]: Please ignore the image itself - we don't have the finals yet

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/taptopay/IsTapToPayAvailable.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.woocommerce.android.ui.payments.taptopay
+
+import com.woocommerce.android.util.FeatureFlag
+import javax.inject.Inject
+
+class IsTapToPayAvailable @Inject constructor() {
+    operator fun invoke() = FeatureFlag.IPP_TAP_TO_PAY.isEnabled()
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/taptopay/IsTapToPayAvailable.kt:7]: Latter on, I will add other conditions here: country, NFC availability etc

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/SelectPaymentMethodFragment.kt [hunk 3]
```diff
@@ -163,7 +166,7 @@ class SelectPaymentMethodFragment : BaseFragment(R.layout.fragment_take_payment)
                         BaseTransientBottomBar.LENGTH_LONG
                     ).show()
                 }
-                is SharePaymentUrl -> {
+                is SelectPaymentMethodViewModel.SharePaymentUrl -> {
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/SelectPaymentMethodFragment.kt]: 🔍 nitpick: We can add an import for `SharePaymentUrl` as we have done for other events.

File: WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            android:layout_height="wrap_content"
+            android:text="@string/card_reader_type_selection_tap_to_pay"
+            android:layout_marginHorizontal="@dimen/major_200"
+            android:layout_marginTop="0dp"
+            app:layout_constraintBottom_toTopOf="@id/reader_selection_use_blueatooth_reader"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/reader_selection_use_blueatooth_reader"
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml]: 🔍 Super nitpick: Sorry about this nitpick. The word Bluetooth is spelled wrong in `reader_selection_use_blueatooth_reader`.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov! I've tested this and it works as expected. I've left couple of nitpicks which are non-blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/SelectPaymentMethodFragment.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/SelectPaymentMethodFragment.kt]: Done! 0d9ea883dbaae7a77f8aafc3e5f4a2f731bb5f2d

File: WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml (hunk 4 above)
Comment (kidinov) [reply WooCommerce/src/main/res/layout/card_reader_type_selection_dialog.xml]: Done - 760521b8b12e56610d1ce14b0a4aa8668d76470e

**Pull request merged**